### DISCUSSION TITLE
will it work for multivariate time series prediction   both regression and classification

### DISCUSSION THREAD

**Discussion by Sandy4321** [blue-yonder/tsfresh#843] [Q&A]
great code thanks
may you clarify : 
will it work for multivariate time series prediction   both regression and classification 
1 
  where all values are continues values 
    weight    height     age   target 
1   56        160    34    1.2
2   77        170    54    3.5
3   87        167     43   0.7
4   55        198    72    0.5
5   88        176     32   2.3

2
or  even will it work for multivariate time series where values are mixture of continues  and categorical values
for example 2 dimensions have continues values and 3 dimensions are categorical values

    color        weight     gender  height  age  target 
1    black       56             m   160    34    yes
2    white       77             f   170    54    no
3    yellow      87             m   167    43    yes
4    white       55             m   198    72    no
5   white       88              f   176    32    yes

**Comment by nils-braun** [comment]
Thanks for your question!
I think I did not 100% understand it though.

I assume you refer to the features selection part. 

Do you refer to the features or the target when you say continuous or categorical?

tsfresh will work for all kind of combinations (feature real or binary, target real or binary). For multi-class selection there exist options for the features selection (which turn this on).

**Comment by Sandy4321** [comment]
yes Feature extraction for further use for  multivariate time series prediction both regression and classification

do you have example where Feature extraction and after it used for multivariate time series prediction both regression and classification

**Comment by nils-braun** [comment]
So just to clarify (because I am still confused concerning your example), `tsfresh` is a time series feature extraction library. It is used to extract features on time series (your data are just "normal" tabular data examples, in which case you probably want to use something similar to https://www.featuretools.com/).
In case you really have a time series use case: yes, we do have examples for prediction with a multivariate method afterwards. For example [this](https://github.com/blue-yonder/tsfresh/blob/main/notebooks/examples/05%20Timeseries%20Forecasting.ipynb), which is a forecasting use case for a regression target. We also have [one](https://github.com/blue-yonder/tsfresh/blob/main/notebooks/examples/02%20sklearn%20Pipeline.ipynb) for classification. And even a more [complex one](https://github.com/blue-yonder/tsfresh/blob/main/notebooks/examples/04%20Multiclass%20Selection%20Example.ipynb) on multiclass feature selection.

**Comment by Sandy4321** [comment]
in tabular data order of rows is not important 
but in my example order of rows is important
consider it like this
each row means the time series, and columns represent different continues or/and discrete/category observation.
 
1 
  where all values are continues values 
         weight    height     age   target 
time 1|   56        160    34        1.2
time 2|   77        170    54        3.5
time 3|   87        167     43       0.7
time 4|   55        198    72        0.5
time 5|   88        176     32       2.3
etc

2
or  even will it work for multivariate time series where values are mixture of continues  and categorical values
for example 2 dimensions have continues values and 3 dimensions are categorical values

			color        weight     gender  height  age      target 
time 1|    black       56             m   160    34         yes
time 2|    white       77             f   170    54         no
time 3|   yellow      87             m   167    43          yes
time 4|    white       55             m   198    72         no
time 5|   white       88              f   176    32         yes

etc

**Comment by nils-braun** [comment]
Ah, now I understand.
* Yes, tsfresh will work for time series prediction with continous values - both for regression and prediction. Just a note: tsfresh is a feature extraction and selection library. So you would need to train a ML method afterwards using those features (and which method you use also depends, if you can have a regression or classification target)
* No, categorical values are not supported by most of the features tsfresh extracts.

**Reply by lynBIT** [reply]
Hi, I have some time series features for regression analysis to model the number of interface calls versus the container CPU, the independent variable is the number of interface calls (two features respectively) and the dependent variable is CPU usage. Both independent and dependent variables are time series data of a sample with an interval of 15s. Now I want to use TsFresh to extract features for regression analysis (other machine learning methods can be used for regression), but I don't know if TsFresh can do that.
this is my dataframe
                    time           features_1         feature_2                cpu
2022-02-21 09:30:00                10                12                    0.106
2022-02-21 09:30:15                12                14                    0.111
2022-02-21 09:30:30                15                16                    0.142
2022-02-21 09:30:45                19                20                    0.142
2022-02-21 09:31:00                30                31                    0.18
2022-02-21 09:31:15                37                23                    0.192
2022-02-21 09:31:30                25                36                    0.193
2022-02-21 09:31:45                35                37                    0.193
2022-02-21 09:32:00                23                39                    0.201
2022-02-21 09:32:15                32                45                    0.211
2022-02-21 09:32:30                33                47                    0.214
2022-02-21 09:32:45                56                48                    0.214
2022-02-21 09:33:00                55                49                    0.215
2022-02-21 09:33:15                45                36                    0.189
2022-02-21 09:33:30                28                36                    0.154
2022-02-21 09:33:45                34                33                    0.154
2022-02-21 09:34:00                23                27                    0.144
2022-02-21 09:34:15                22                24                    0.142
2022-02-21 09:34:30                21                18                    0.137
2022-02-21 09:34:45                16                15                    0.134
